util/fpga/splice_rom.sh

```diff
@@ -20,16 +20,84 @@ set -e
 
 . util/build_consts.sh
 
+function usage() {
+  cat << USAGE
+Utility script to load ROM contents into the FPGA bitstream.
+
+Usage: $0 [-t TARGET_BOARD] [-T TARGET_TOP] [-b DV | PROD]
+
+  - t: Target board: cw310, nexysvideo.
+  - T: Target top: earlgrey.
+  - b: Mask ROM binary, set to either DV or PROD.
+
+Mask ROM binary targets (-b):
+  - DV: sw/device/boot_rom
+  - PROD: sw/device/silicon_creator/mask_rom
+
+USAGE
+}
+
 # Change these variables when using the script for a different top level and/or FPGA board.
-TARGET_BOARD="cw310"
-TARGET_TOP="earlgrey"
-TARGET_FILE_EXT=".scr.39.vmem"
+FLAGS_TARGET_BOARD="cw310"
+FLAGS_TARGET_TOP="earlgrey"
+FLAGS_BIN="DV"
+
+# `getopts` usage
+# - The initial colon in the optstring is to suppress the default error
+#   handling.
+# - The remaining options are specified in alphabetical order, and the case
+#   statement should match this order.
+# - Only options that take an argument should have a following colon.
+# - The case statement contains two additional cases:
+#   - when `$flag` = `?`, this is an unexpected option.
+#   - when `$flag` = `:`, this is the case that a flag which requires an
+#     argument is not provided one. In both cases, `$OPTARG` contains the
+#     relevant parsed option.
+# - After option parsing is finished, we `shift` by `$OPTIND - 1` so that the
+#   remaining (unprocessed) arguments are in `$@` (and $1, $2, $3 etc.).
+while getopts ':b:t:T:' flag; do
+  case "${flag}" in
+    b) FLAGS_BIN="${OPTARG}";;
+    t) FLAGS_TARGET_BOARD="${OPTARG}";;
+    T) FLAGS_TARGET_TOP="${OPTARG}";;
+    \?) echo "Unexpected option: -${OPTARG}" >&2
+        usage
+        exit 1
+        ;;
+    :) echo "Option -${OPTARG} requires a path argument" >&2
```

Comment:
I'm not sure "path argument" is right here (for example, it doesn't apply to `-t`)

Reply:
Fixed. Thanks for the review!